### PR TITLE
Offline Handling

### LINKED ISSUE #382: Offline mode handling

### PR DESCRIPTION
**Pull request by mathieumognol**
Close #382 

- Created `ConnectivityStatus`  that allows to know if the user is online or offline throughout the app
- Created a component `CoonectivityWrapper` that manages UI when user is offline
- Removed access to the flight screen when offline
- Remove possibility to try to add flight or add user when offline
- Remove possibility to try to send message when offline

### CODE REVIEW

**Review by Plouc314 — APPROVED** [review]
Looking good, it is a bit tedious to have to explicitly have to pass it as argument everywhere but I guess it is the most straightforward approach (I'm guessing there has to be some equivalent to React's [useContext](https://react.dev/reference/react/useContext) but I'm not familiar).
~~Also you might want to add some tests~~ (I didn't see it is still a draft)

File: app/src/main/java/ch/epfl/skysync/screens/crewpilot/Flight.kt [hunk 1]
```diff
@@ -72,8 +73,10 @@ fun UserLocationMarker(location: Location, user: User) {
 fun FlightScreen(
     navController: NavHostController,
     inFlightViewModel: InFlightViewModel,
-    uid: String
+    uid: String,
+    connectivityStatus: ConnectivityStatus
```
[the remark this answers is not in the document]
Comment (mathieumognol) [reply app/src/main/java/ch/epfl/skysync/screens/crewpilot/Flight.kt:77]: Nope forgot to delete it my bad

File: app/src/main/java/ch/epfl/skysync/components/ChatText.kt [hunk 2]
```diff
@@ -54,10 +54,17 @@ import ch.epfl.skysync.ui.theme.veryLightGreen
...
+fun ChatText(
+    messages: List<ChatMessage>,
+    onSend: (String) -> Unit,
+    paddingValues: PaddingValues,
+    connectivityStatus: ConnectivityStatus
+) {
   Column(modifier = Modifier.padding(paddingValues).imePadding()) {
     ChatTextBody(messages, modifier = Modifier.weight(1f))
-    ChatInput(onSend)
+    if (connectivityStatus.isOnline()) {
```
Comment (Prolas) [inline app/src/main/java/ch/epfl/skysync/components/ChatText.kt:65]: Instead of displaying the button if online you could disable the button when offline

Comment (mathieumognol) [reply app/src/main/java/ch/epfl/skysync/components/ChatText.kt:65]: Good to know.
Thanks ;)

File: app/src/main/java/ch/epfl/skysync/components/ConnectivityStatus.kt [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import android.content.Context
+import android.net.ConnectivityManager
+
+class ConnectivityStatus(private val context: Context) {
+
+  fun isOnline(): Boolean {
+    val connectivityManager =
+        context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+    val activeNetwork = connectivityManager.activeNetworkInfo
+    return activeNetwork != null && activeNetwork.isConnected
```
[the remark this answers is not in the document]
Comment (mathieumognol) [reply app/src/main/java/ch/epfl/skysync/components/ConnectivityStatus.kt:16]: Works great thanks

**Pull request merged**